src/PlugWallet/index.ts

```diff
@@ -150,7 +153,37 @@ class PlugWallet {
   };
 
   public getTransactions = async (): Promise<GetTransactionsResponse> => {
-    return getTransactions(this.accountId);
+    const { secretKey } = this.identity.getKeyPair();
+    const icpTransactions = await getICPTransactions(this.accountId);
+    const tokenTransactions = await Promise.all(
+      this.registeredTokens
+        .filter(token => token.features.includes('history'))
+        .map(async token => {
+          const tokenActor = await createTokenActor(
+            token.canisterId,
+            secretKey
+          );
+          console.log('fetching history for', token.symbol);
+          const history = await tokenActor.events({
+            after: [BigInt(512)],
+            limit: 1000,
+          });
+          console.log('all trx', history, history.data.length);
+          return history.data.filter(
+            val =>
+              ('Transfer' in val.kind &&
+                (val.kind.Transfer.from.toText() === this.principal ||
+                  val.kind.Transfer.to.toText() === this.principal)) ||
+              ('Mint' in val.kind &&
+                val.kind.Mint.to.toText() === this.principal) ||
+              ('Burn' in val.kind &&
+                (val.kind.Burn.from.toText() === this.principal ||
+                  val.kind.Burn.to.toText() === this.principal))
+          );
+        })
```

Comment:
@qti3e do you know why this might be empty? disregard the `after` and `limit`. 
No matter what combination I try I don't seem to find my trxs. Any clues?

Reply:
cc @Nima-Ra